crates/wit-component/tests/interfaces/reference-out-of-order/world.wit

```diff
@@ -3,14 +3,14 @@ interface foo {
     s: string,
   }
 
-  record r-no-string {
-    s: u32,
-  }
-
   variant v {
     s(string),
   }
 
+  record r-no-string {
```

Comment:
Note to self / other reviewers: the order here changed because encoding now encodes types as encountered in function signatures rather than first passing over the types in the interface to encode. Thus it's `r`, `v`, `r-no-string`, `v-no-string` now.